### ISSUE TITLE
v1.0.2 to-do

### ISSUE THREAD

**Issue by JacksonBurns** [JacksonBurns/fastprop#12]
- Remove this duplicated line:
 https://github.com/JacksonBurns/fastprop/blob/0ab3dd7e02c955bff3e24db85c814681b6b86d1f/fastprop/cli/train.py#L36
 - fix the `get_metrics` function in `model.fastprop` to return the new names and update `hopt.py` to use the correct metric during optimization

**Comment by JacksonBurns** [comment]
- fill in missing return type docstring in `train_and_test` and allow disabling the progress bar

**Comment by JacksonBurns** [comment]
- change the `with_resources` setup in hyperparameter optimization to this: `resources={"gpu": 1, "cpu": psutil.cpu_count()}`

**Comment by JacksonBurns** [comment]
- hyperparameter optimization should choose the optimal model based on the validation performance, not the test performance

**Comment by JacksonBurns** [comment]
- hyperparameter optimization should hold out a separate test set at the beggining instead of striping it through the dataset along with training and validation

**Comment by JacksonBurns** [comment]
- return the SMILES strings from clean dataset for passing to `split`

**Comment by JacksonBurns** [comment]
- make the output folder creation safe for DDP
